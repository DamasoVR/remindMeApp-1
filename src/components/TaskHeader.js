/**
 * RemindMe App
 * TasksScreen
 */

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableHighlight,
  Image,
  Dimensions,
  ImageBackground,
  View
} from 'react-native';
import Colors from './../theme/colors';
import Typografy from './../theme/typografy';

const screen = Dimensions.get('window');


export default class TaskHeader extends Component<{}> {

  getFormatedDate(){
    //TODO: Move this to a static data folder
    const days = [ 'Lunes', 'Martes', 'Miércoles', 'Jueves', 'Viernes', 'Sábado', 'Domingo'];
    const months = ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre'];

    const currentTime = new Date(),
          month = months[currentTime.getMonth()],
          day = days[currentTime.getDay()],
          date = currentTime.getDate(),
          year = currentTime.getFullYear();

    return `${day} ${date} de ${month} de ${year}`.toUpperCase();
  }

  render() {

    const { taksToBeCompleted } = this.props;

    return (
      <View style={styles.container}>
        <ImageBackground
          style={styles.containerBackground}
          source={require('./../images/tasks-bg.jpg')}>
          <Image
            style={styles.avatar}
            source={require('./../images/user-avatar.jpg')}/>
          <Text style={styles.todosText}>{taksToBeCompleted} Pendientes</Text>
          <Text style={styles.dateText}>{this.getFormatedDate()}</Text>
        </ImageBackground>
      </View>
    );
  }
}

// TODO: Create Theme styles (Buttons, Global Components, etc) so it can be imported on each component as required.
const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: screen.width,
    height: 399,
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  containerBackground: {
    flex: 1,
    justifyContent: 'center',
    width: screen.width,
    height: 399,
    alignItems: 'center'
  },
  avatar:{
    width: 130,
    height: 130,
    borderRadius: 65
  },
  todosText:{
    ...Typografy.avenirFont,
    fontSize: 36,
    color: Colors.white
  },
  dateText:{
    ...Typografy.avenirFont,
    fontSize: 16,
    color: Colors.gray,
    fontFamily: 'Avenir'
  }
});
