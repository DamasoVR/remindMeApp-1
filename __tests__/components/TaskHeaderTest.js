import 'react-native';
import React from 'react';
import TaskHeader from '../../src/components/TaskHeader';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

//Mock Data
const MockDate = require('mockdate');
      MockDate.set('2018-01-10T12:00:00'),
      taskToBeCompleted = 10,
      TaskHeaderClass = new TaskHeader({taskToBeCompleted});

describe('Task Header Component - ', () => {

  afterAll( () => {
    MockDate.reset();
  });

  it('should render correctly', () => {
    const component = renderer.create(
      <TaskHeader taksToBeCompleted={ taskToBeCompleted }/>
    ).toJSON();
    expect(component).toMatchSnapshot();
    expect(TaskHeaderClass.props.taskToBeCompleted).toBe(taskToBeCompleted);
  });

  it('should generate date with format', () => {
    expect(TaskHeaderClass.getFormatedDate()).toBe('JUEVES 10 DE ENERO DE 2018');
  });

});
