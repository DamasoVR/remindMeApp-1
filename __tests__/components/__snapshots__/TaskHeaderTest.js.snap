// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Header Component -  should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": 399,
      "justifyContent": "center",
      "overflow": "hidden",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 399,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 399,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "borderRadius": 65,
          "height": 130,
          "width": 130,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgb(255,255,255)",
          "fontFamily": "Avenir",
          "fontSize": 36,
        }
      }
    >
      10
       Pendientes
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgb(164, 164, 164)",
          "fontFamily": "Avenir",
          "fontSize": 16,
        }
      }
    >
      JUEVES 10 DE ENERO DE 2018
    </Text>
  </View>
</View>
`;
